lib/ansible/modules/storage/netapp/na_ontap_qtree.py

```diff
@@ -110,107 +135,151 @@ def __init__(self):
         query_details = netapp_utils.zapi.NaElement.create_node_with_children(
-            'qtree-info', **{'vserver': self.vserver,
-                             'volume': self.flexvol_name,
+            'qtree-info', **{'vserver': self.parameters['vserver'],
+                             'volume': self.parameters['flexvol_name'],
                              'qtree': name})
-
         query = netapp_utils.zapi.NaElement('query')
         query.add_child_elem(query_details)
         qtree_list_iter.add_child_elem(query)
-
         result = self.server.invoke_successfully(qtree_list_iter,
                                                  enable_tunneling=True)
-
+        return_q = False
         if (result.get_child_by_name('num-records') and
                 int(result.get_child_content('num-records')) >= 1):
-            return True
-        else:
-            return False
+            return_q = {'export_policy': result['attributes-list']['qtree-info']['export-policy'],
+                        'unix_permissions': result['attributes-list']['qtree-info']['mode'],
+                        'oplocks': result['attributes-list']['qtree-info']['oplocks'],
+                        'security_style': result['attributes-list']['qtree-info']['security-style']}
+
+        return return_q
 
     def create_qtree(self):
+        """
+        Create a qtree
+        """
+        options = {'qtree': self.parameters['name'], 'volume': self.parameters['flexvol_name']}
+        if self.parameters.get('export_policy'):
+            options['export-policy'] = self.parameters['export_policy']
+        if self.parameters.get('security_style'):
+            options['security-style'] = self.parameters['security_style']
+        if self.parameters.get('oplocks'):
+            options['oplocks'] = self.parameters['oplocks']
+        if self.parameters.get('unix_permissions'):
+            options['mode'] = self.parameters['unix_permissions']
         qtree_create = netapp_utils.zapi.NaElement.create_node_with_children(
-            'qtree-create', **{'volume': self.flexvol_name,
-                               'qtree': self.name})
-
+            'qtree-create', **options)
         try:
             self.server.invoke_successfully(qtree_create,
                                             enable_tunneling=True)
-        except netapp_utils.zapi.NaApiError as e:
-            self.module.fail_json(msg="Error provisioning qtree %s: %s" % (self.name, to_native(e)),
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg="Error provisioning qtree %s: %s"
+                                  % (self.parameters['name'], to_native(error)),
                                   exception=traceback.format_exc())
 
     def delete_qtree(self):
-        path = '/vol/%s/%s' % (self.flexvol_name, self.name)
+        """
+        Delete a qtree
+        """
+        path = '/vol/%s/%s' % (self.parameters['flexvol_name'], self.parameters['name'])
```

Comment:
Not sure this would work if flexvol_name is not specified.

Reply:
is the same code that was before. Just using na_helper.set_parameters function to fill parameters
flexvol_name is a required module option ('name'). Not sure what do you mean